### PR TITLE
Port async job supervision to Windows

### LINKED ISSUE #67: Port async job supervision to Windows
## Summary

Job supervision (the async `agy_run` / `agy_status` / `agy_cancel` tools) is currently Linux-only. `StartJob` and `supervisor.Run` refuse early via `proc.Supported` (`internal/proc/proc_other.go`), so on Windows only the synchronous path works: `agy_run_sync`, `list_models`, `list_sessions`.

This issue tracks porting job supervision to Windows. The codebase is already structured for it — OS-specific primitives are isolated behind build tags (`_linux.go` / `_other.go`), and all authoritative state is persisted to disk (`meta.json`, the `exit_code` sentinel). Every Linux mechanism has a well-established Win32 analog available via `golang.org/x/sys/windows`. **There is no hard blocker.**

## Linux mechanisms → Windows equivalents

| Mechanism (Linux) | Location | Windows equivalent | Notes |
|---|---|---|---|
| Process group to kill agy + all descendants (`Setpgid`, `Kill(-pid, sig)`) | `internal/proc/proc_linux.go` | **Job Objects** — `CreateJobObject` + `AssignProcessToJobObject` + `TerminateJobObject` (optionally `JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE`) | Canonical Windows way to kill a process tree; as reliable as / better than pgroups |
| `flock` cross-process key lock | `internal/manager/keylock_linux.go` | `LockFileEx` with `LOCKFILE_EXCLUSIVE_LOCK \| LOCKFILE_FAIL_IMMEDIATELY` | Near 1:1 replacement |
| `/proc` liveness — boot_id + PID + starttime + comm (PID-recycling defense) | `internal/manager/liveness_linux.go` | `OpenProcess` + `GetProcessTimes` (**creation time** = exact analog of starttime) + `QueryFullProcessImageName` (comm analog) | Creation-time pinning subsumes boot_id: a PID recycled after reboot has a different creation time, so the boot_id concept can be dropped on Windows |
| Detached supervisor that outlives the manager (`Setpgid` + detached stdio + init adoption) | `internal/manager/manager.go` `StartJob` | Creation flags `DETACHED_PROCESS` / `CREATE_NO_WINDOW` + `CREATE_BREAKAWAY_FROM_JOB` | No init-reaper on Windows, but also no zombies; the `exit_code` disk sentinel already makes the restore path work when the manager dies. `cmd.Wait` is only an optimization. |
| Signal cancel/timeout (SIGTERM → 10s grace → SIGKILL) | `internal/supervisor/supervisor.go`, `internal/manager/cancel.go` | See "Design decisions" below | The genuinely lossy part |

`os.DevNull` (agy's stdin redirect) is already cross-platform (`NUL` on Windows) — no change needed.

## Proposed file layout

Fill in new `_windows.go` implementations rather than rewriting logic. Currently the non-Linux files are tagged `//go:build !linux`; these need to split so Windows gets real implementations while remaining unsupported platforms keep the stubs.

- `internal/proc/proc_windows.go` — Job Object create/assign/terminate; `Supported = true`.
- `internal/manager/liveness_windows.go` — `OpenProcess` + `GetProcessTimes` + `QueryFullProcessImageName`.
- `internal/manager/keylock_windows.go` — `LockFileEx` / `UnlockFileEx`.
- `internal/supervisor/signal_windows.go` — exit-code mapping (mostly trivial, see below).
- Cancel channel for manager → supervisor (see below).
- Retag `proc_other.go` / `liveness_other.go` / `keylock_other.go` / `signal_other.go` from `!linux` to `!linux,!windows` so they remain the stubs for other platforms.

## Design decisions to make

1. **Manager → supervisor cancel channel.** Linux does `Kill(supervisorPID, SIGTERM)`. Windows has no POSIX signals for arbitrary processes. Two options:
   - **`cancel` sentinel file** in the job dir that the supervisor polls — fits the existing "state on disk" design, dead simple, fully cross-platform. *(Recommended.)*
   - **Per-job named event** (`CreateEvent` / `SetEvent`, supervisor `WaitForSingleObject`) — lower latency, more Win32 machinery.

2. **Graceful cancel is the real semantic loss.** Linux gives agy a `SIGTERM` + 10s flush window before `SIGKILL`. `TerminateJobObject` is a hard kill — no flush window. Options:
   - Ship hard-kill-only cancel/timeout on Windows (document the difference). *(Simplest.)*
   - Attempt a soft `GenerateConsoleCtrlEvent(CTRL_BREAK_EVENT)` first, then hard-kill after grace — only works if agy is a console app in its own process group that handles the event (unverified for agy).

3. **Exit-code signal decoding is mostly moot on Windows.** `signalExitCode` (128+signal) exists because Linux learns *why* agy died from its wait status. On Windows the supervisor already owns the `timedOut` / `cancelled` channels and `resolveExitCode` overrides on those flags, so `signal_windows.go` is nearly trivial.

## Testing

This is likely the larger half of the work. There is a substantial `*_linux_test.go` suite (admission, cancel, capture, gc, liveness, restore) plus `internal/testutil/fakesupervisor.go`, which literally writes to `/proc/$$/comm`. A port needs:
- A Windows `fakesupervisor` equivalent (set a distinct image name instead of writing `comm`).
- Windows analogs of the liveness / cancel / restore tests.

## Effort estimate

Production code: **moderate** — ~4–5 new `_windows.go` files, a few hundred lines, every primitive has a clear Win32 analog. Cost is dominated by (a) the fiddly detachment / breakaway edge cases (esp. when the MCP host already placed `agy-mcp` in a job object) and (b) test parity — not the core primitives.

## Acceptance criteria

- [ ] `agy_run`, `agy_status`, `agy_cancel` work on Windows.
- [ ] Cancel and hard timeout reliably terminate agy **and all descendants** (Job Object).
- [ ] Cross-process key serialization works between two `agy-mcp` processes on Windows (`LockFileEx`).
- [ ] Liveness correctly rejects a recycled PID (creation-time mismatch).
- [ ] A restored job whose supervisor outlived a manager restart is re-tracked; a manager crash still yields a correct terminal state from the `exit_code` sentinel.
- [ ] Windows behavioral differences (notably non-graceful cancel, if chosen) are documented in the README.
- [ ] Windows test coverage for the ported paths.

---
*Filed from analysis of the current `main` branch; no code changed. Implementation to follow in a separate session.*

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Job supervision (the async `agy_run` / `agy_status` / `agy_cancel` tools) was Linux-only: `StartJob` and `supervisor.Run` refused early via `proc.Supported`. This implements it on Windows behind the existing `_linux.go` / `_windows.go` / `_other.go` build-tag split, so async jobs now work on Windows while macOS keeps the clear "unsupported" error. Linux behavior is unchanged.

Every Linux primitive maps to a Win32 analog:

| Mechanism (Linux) | Windows equivalent |
|---|---|
| Process group kill (`Setpgid`, `kill -pgid`) | **Job Objects** (`CreateJobObject` + `AssignProcessToJobObject` + `TerminateJobObject`) |
| `flock` cross-process key lock | `LockFileEx` (`LOCKFILE_EXCLUSIVE_LOCK \| LOCKFILE_FAIL_IMMEDIATELY`) |
| `/proc` liveness (boot_id + pid + starttime + comm) | `OpenProcess` + `GetProcessTimes` creation time + `WaitForSingleObject` + `QueryFullProcessImageName` |
| Detached supervisor (`Setpgid`, init adoption) | `DETACHED_PROCESS` + `CREATE_BREAKAWAY_FROM_JOB` (when the parent job permits it) |
| Cancel via `SIGTERM` to the supervisor | a `cancel` sentinel file the supervisor polls for |

### Key changes

- **`proc`**: replaced the pid-based `SetGroup` / `TerminateGroup(pid, sig)` with a handle-based `Group` API (`ConfigureGroup`, `StartDetached`, `Track`, `Group.Terminate`, `Group.Close`). Linux still uses process groups (behavior-preserving); Windows uses Job Objects. The supervisor tracks agy with **kill-on-close**, so if the supervisor process dies unexpectedly agy and its descendants are torn down too (stricter than the Linux orphan-and-reparent baseline). `StartDetached` adds `CREATE_BREAKAWAY_FROM_JOB` only after confirming the parent job permits breakaway.
- **Liveness (Windows)**: process creation time (`GetProcessTimes`) is an absolute wall-clock value, so it pins a PID across reboots and subsumes the Linux boot-id defense; `WaitForSingleObject` confirms the process is still running, with an image-name fallback for older metas.
- **Cancel**: Windows has no signal for an arbitrary process, so the manager writes a `cancel` sentinel into the job dir and the supervisor polls for it; Linux keeps `SIGTERM`. **Windows difference:** cancel and hard-timeout are hard kills (no `SIGTERM` grace window).
- **Cross-process key lock (Windows)**: `LockFileEx`, released on handle close, matching the Linux flock model.
- Retagged the `_other.go` stubs from `!linux` to `!linux && !windows`; added `signal_windows.go`.
- **CI**: the test suite now runs on `windows-latest`. **README**: documents the two Windows behavioral differences.

## Related Issues

Closes #67

## Test Plan

- [x] Builds and vets on Linux, macOS, and Windows (`GOOS={linux,darwin,windows} go build/vet ./...`)
- [x] Full test suite green on Linux (`go test ./... -race`), no regressions
- [x] Full test suite green on a Windows 11 host, covering the ported paths:
  - Job Object tree-kill and kill-on-close
  - hard-timeout termination of agy + its child tree (`ExitTimeout`)
  - cancel-via-sentinel (`ExitSIGTERM`)
  - creation-time liveness (live / dead / recycled-PID mismatch / image-name fallback)
  - `LockFileEx` cross-process contention and release-on-close
- [x] Unix-shell-based tests (fake agy/supervisor scripts) tagged `!windows`; the ported paths get dedicated `_windows_test.go` coverage plus a Windows supervisor integration test driving a fake agy via `cmd.exe`

### Known Windows differences (documented in the README)

- Cancel and hard timeout are **hard kills** (`TerminateJobObject`), with no `SIGTERM` flush window.
- Cancel is delivered via a **sentinel file**, not a signal.
- The assign-after-`Start` window (a grandchild spawned in the microseconds before `AssignProcessToJobObject`) is an accepted risk inherent to Go's `os/exec` (no `CREATE_SUSPENDED` + resume); agy spends its startup initializing before spawning children, so it is not a practical concern.

## Summary by CodeRabbit

* **New Features**
  * Added Windows support for job supervision, including process tracking, cancellation, timeout handling, and liveness checks.
  * Supervisor cancellation now works via a sentinel file on Windows, while Linux continues using signal-based cancellation.
  * Expanded CI to run tests on Windows and added Windows-specific coverage.

* **Bug Fixes**
  * Improved process cleanup and termination behavior across platforms.
  * Updated unsupported-platform handling and documentation to reflect Linux, Windows, and macOS behavior more clearly.

### CODE REVIEW

File: internal/proc/proc_windows.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+// ignored) or in a job that explicitly permits breakaway, and false when it is in
+// a job that forbids it (where the flag would make CreateProcess fail). It fails
+// safe toward not setting the flag when the job cannot be inspected. x/sys/windows
+// does not wrap IsProcessInJob, so that one call is made through a lazy proc.
+func canBreakaway() bool {
+	var inJob int32
+	r1, _, _ := procIsProcessInJob.Call(uintptr(windows.CurrentProcess()), 0, uintptr(unsafe.Pointer(&inJob)))
+	if r1 == 0 {
+		return true // IsProcessInJob failed; if not in a job the flag is harmless
+	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/proc/proc_windows.go:173]: Good catch, fixed in 1026dcf. `canBreakaway` now returns false when `IsProcessInJob` itself fails (r1 == 0), so a failed probe never sets CREATE_BREAKAWAY_FROM_JOB and can't cause a CreateProcess spawn failure in a breakaway-forbidding parent job. This also matches the function's docstring (fail safe toward not setting the flag).

File: internal/manager/cancel_windows.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+func (m *Manager) requestCancel(meta jobstore.Meta) error {
+	dir, err := m.store.Dir(meta.ID)
+	if err != nil {
+		return err
+	}
+	f, err := os.OpenFile(jobstore.CancelPath(dir), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o600)
+	if err != nil {
+		return err
+	}
+	return f.Close()
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/manager/cancel_windows.go]: Done in 1026dcf. The supervisor only stats the sentinel for existence (an empty file is the valid signal), so a torn read wasn't reachable here, but I made the write atomic (temp + rename) anyway to match the meta.json / exit_code job-dir contract and future-proof against a content-reading consumer.

**Comment by tphakala** [comment]
@coderabbitai review
